src/webgl/texture.js

```diff
@@ -344,47 +344,77 @@ export default class Texture extends Resource {
     ({type, dataFormat, compressed, width, height} = this._deduceParameters({
       format, type, dataFormat, compressed, data, width, height}));
 
-    // Support ndarrays
-    if (data && data.data) {
-      const ndarray = data;
-      data = ndarray.data;
-      width = ndarray.shape[0];
-      height = ndarray.shape[1];
+    const {gl} = this;
+    gl.bindTexture(this.target, this.handle);
```

Comment:
Yes we should be consistent here. There are four subclasses of texture and they need different targets, that is what we need to accommodate. Let's review after landing as you suggest.